src/Data/Form.purs

```diff
@@ -176,50 +254,91 @@ eval get page = do
       placeholder <- evalExpr get input.placeholder
       required <- evalExpr get input.required
       value <- traverse (evalExpr get) input.value
-      pure
-        ( Currency
-          { default
-          , placeholder
-          , required
-          , value
-          }
-        )
+      pure $ validate $ Currency
+        { default
+        , placeholder
+        , required
+        , value
+        , errors: mempty
+        }
```

Comment:
So right now I construct a valid `Input ExprType`, then run it through the `validate` function which adds on validation. My issue with this is that the compiler doesn't enforce validation. You could easily forget to add the call to `validate` here and the compiler wouldn't complain. I'm wondering if we should/could add another `f` that represents a validated form, and have `errors` only be a label on `Input`s for that `f` if that's possible. Something like

```purescript
toggleExpr :: Input Expr
toggleExpr = Toggle
  { default: Just (Val $ Boolean true)
  , value: NotSet
  }

toggleExprType :: Input ExprType
toggleExprType = Toggle
  { default: Just (Boolean true)
  , value: NotSet
  }

toggleValidated :: Input Validated
toggleValidated = Toggle
  { default: Just (Boolean true)
  , value: NotSet
  , errors: mempty
  }
```

Reply:
Yeah, I feel similarly about the lack of static checking for errors. I made some comments about this: https://github.com/citizennet/purescript-lynx/pull/16#discussion_r268286652.

Ultimately, I think we need to decide what `Invalid` means for `InputSource _`. If it's supposed to represent an invalid value, it seems like that's where we should be putting errors. What do you think?